src/scene/vcPOI.cpp

```diff
@@ -69,11 +358,34 @@ vcPOI::vcPOI(vdkProject *pProject, vdkProjectNode *pNode, vcState *pProgramState
 
   memset(&m_flyThrough, 0, sizeof(m_flyThrough));
 
+  switch (pProgramState->activeTool)
```

Comment:
Similiarly, this cannot be here- this should be a param or something otherwise if a update happens over the network while the user is in a tool this will break.

Reply:
Roger

Reply:
I have moved this to OnNodeUpdate(), although I don't quite see the difference if OnNodeUpdate is called in the constructor?

Reply:
OnNodeUpdate is called whenever the node changes (this user or another user)

Reply:
Right, although we already `UpdateState()` in OnNodeUpdate. This switch statement simply initialises state in the constructor. I've moved this to OnNodeUpdate, but it will only ever be hit once; during construction.